### PR TITLE
Fix #17472 Skip validation annotations for response and fix validation annotations for list arguments

### LINKED ISSUE #17472: [BUG] generated code cannot be compiled (Java 21)
plugin version 7.2.0 generated code like below.

There are 2 problems:
1. code is not valid Java code (causes compiler error)
2. min and max length is not restriction for list itself, but for individual items in the list

Is there a workaround, how to switch this behavior off? ==> Possibly yes, not sure it will not disable.other older features -> `<useBeanValidation>false</useBeanValidation>`
(it was introduced here: https://github.com/OpenAPITools/openapi-generator/commit/809b3331a95b3c3b7bcf025d16ae09dc0682cd69)

```
class ExamplesApi.java
    default ResponseEntity<List<(min = 2, max = 2)String>> getLocalizations(

    //original code generated by 7.1.0:
    // default ResponseEntity<List<String>> getLocalizations(

class ExamplesApiDelegate.java
    default ResponseEntity<List<(min = 2, max = 2)String>> getLocalizations(
    //original code generated by 7.1.0:
    // default ResponseEntity<List<String>> getLocalizations(
```

example-service.json:
```
{
    "openapi" : "3.0.0",
    "info" : {
        "title" : "example-service",
        "description" : "undefined",
        "version" : "1.0.0"
    },
    "paths" : {
        "/localizations" : {
            "get" : {
                "description" : "example",
                "operationId" : "getLocalizations",
                "responses" : {
                    "200" : {
                        "description" : "OK",
                        "content" : {
                            "application/json" : {
                                "schema" : {
                                    "type" : "array",
                                    "items" : {
                                        "maxLength" : 2,
                                        "minLength" : 2,
                                        "type" : "string"
                                    }
                                }
                            }
                        }
                    }
                },
                "parameters" : [
                    {
                        "in" : "query",
                        "name" : "X-Correlation-ID",
                        "required" : false,
                        "schema" : {
                            "type" : "string"
                        }
                    },
                    {
                        "in" : "query",
                        "name" : "X-Message-ID",
                        "required" : false,
                        "schema" : {
                            "type" : "string"
                        }
                    }
                ],
                "tags" : [
                    "Examples"
                ]
            }
        }
    },
    "servers" : [
        {
            "url" : "http://localhost:8080/"
        }
    ],
    "tags" : [
        {
            "name" : "Examples",
            "description" : "Examples operations"
        }
    ]
}
```
plugin config

```
                <plugin>
                    <groupId>org.openapitools</groupId>
                    <artifactId>openapi-generator-maven-plugin</artifactId>
                    <version>7.2.0</version>
                    <executions>
                        <execution>
                            <goals>
                                <goal>generate</goal>
                            </goals>
                        </execution>
                    </executions>
                    <configuration>
                        <output>${project.basedir}/src/main/openapi</output>
                        <inputSpec>example-service.json</inputSpec>
                        <generatorName>spring</generatorName>
                        <library>spring-boot</library>
                        <ignoreFileOverride>.openapi-generator-ignore</ignoreFileOverride>
                        <supportingFilesToGenerate>OpenAPI2SpringBoot.java,OpenApiGeneratorApplication.java,RFC3339DateFormat.java,ApiUtil.java,</supportingFilesToGenerate>
                        <importMappings>SpecialId=org.example.common.SpecialId,</importMappings>
                        <typeMappings>special-id=SpecialId</typeMappings>
                        <configOptions>
                            <useSpringBoot3>true</useSpringBoot3>
                            <configPackage>org.example.service.configuration</configPackage>
                            <apiPackage>org.example.service.api</apiPackage>
                            <modelPackage>org.example.service.model</modelPackage>
                            <invokerPackage>org.example</invokerPackage>
                            <delegatePattern>true</delegatePattern>
                            <swaggerDocketConfig>false</swaggerDocketConfig>
                            <returnSuccessCode>true</returnSuccessCode>
                            <useTags>true</useTags>
                            <dateLibrary>java8</dateLibrary>
                            <hideGenerationTimestamp>true</hideGenerationTimestamp>
                        </configOptions>
                    </configuration>
                </plugin>

```

### PR DESCRIPTION
**Pull request by jpfinne**
Since [#4947] [java]: adds support for validation of primitives in arrays, the generated java code can be uncompilable. 

 See 
[BUG] generated code cannot be compiled (Java 21) #17472
[BUG][SPRING] Webflux minItems generates incompatible validation annotation #17564

The issue is in the regex in 
postProcessResponseWithProperty()
property.dataType = property.dataType.replaceAll("(?:(?i)@[a-z0-9]*+\\s*)*+", "");

The regex only strips simple `@Valid` but fails with `@Size(min=10)`.  
So gives List<@Size(min=10)String> is converted into List<(min=10)String> instead of List<String>.
It seems too complex to fix the regex especially for @Pattern.

This pull request takes a different approach.  It does not generate the bean validation for response so there is not need to clean the response in the post processor.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the 7.3.0
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@wing328 you did extensive review on the code of @Aliaksie.  
Can you review this pull request?

@cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02) @Zomzog (2022/09) @martin-mfg (2023/08)
Please review

### CODE REVIEW

**Comment by wing328** [comment]
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/OpenAPITools/openapi-generator/graphs/contributors.

Let me know if you need help fixing it.

Ref: https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-can-i-update-commits-that-are-not-linked-to-my-github-account

**Comment by MelleD** [comment]
A test would be great

**Comment by jpfinne** [comment]
@MelleD  Test added.
@wing328  GitHub account set (I hope so)

**Comment by MelleD** [comment]
@jpfinne Thanks for the tests :).

I think a few more test cases are necessary for good coverage. Take a look at the other BeanValidation test, maybe you can expand that too. There are a lot of combinations.

e.g.
- what happens to a Set
- What happens if the list has a Dto class
- What about other validations regex etc.

**Comment by jpfinne** [comment]
@MelleD a bit overkill when knowing the implementation.
But let's imagine we want blackbox testing, it it can speed up the merge...

I've tested your 3 uses cases:

- what happens to a Set
- What happens if the list has a Dto class
- What about other validations regex etc.

ps: There is no space between `@Valid `and `Size `in the generated `@Valid@Size(min = 5) `.  This is valid java code!
Something already present in 7.1.0
I've added JavaFileAssert.fileContainsPattern so that the test stays valid for `@Valid@Size(min = 5) ` and for `@Valid @Size(min = 5) `.

**Review by martin-mfg — CHANGES_REQUESTED** [review]

File: modules/openapi-generator/src/test/resources/bugs/issue_17472.yaml
[the remark this answers is not in the document]
Comment (jpfinne) [reply modules/openapi-generator/src/test/resources/bugs/issue_17472.yaml:1]: @martin-mfg Thanks for review.
Indeed the contract was corrupted.

**Comment by jpfinne** [comment]
@martin-mfg The contract is updated

**Comment by MelleD** [comment]
> a bit overkill when knowing the implementation.

Why? There is already a test and file with some combination etc.
Why you don’t reuse it?

https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/test/resources/bugs/issue_4947.yaml

**Review by martin-mfg — APPROVED** [review]

**Comment by wing328** [comment]
can you please resolve the merge conflicts when you've time?

**Comment by wing328** [comment]
closed via https://github.com/OpenAPITools/openapi-generator/pull/19093

**Pull request closed without merging**